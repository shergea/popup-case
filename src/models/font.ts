export default interface Font {
  entities: Array<FontItem>;
  loading: "idle" | "pending" | "succeeded" | "failed";
  expireTime: number;
}

export interface FontItem {
  family: string;
  subsets: string[];
  version: string;
  category: string;
  variants: string[];
}
