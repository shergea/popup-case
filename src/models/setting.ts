export default interface Setting {
  headline: string;
  description: string;
  successMessage: string;
  buttonText: string;
}
