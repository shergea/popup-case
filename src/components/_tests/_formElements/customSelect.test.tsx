import { Formik } from "formik";
import renderer from "react-test-renderer";
import { FontItem } from "../../../models/font";
import CustomSelect from "../../formElements/customSelect";

it("renders correctly", () => {
  const fieldName = "firstName";
  const labelName = "First Name";
  const fonts = [
    {
      family: "ABeeZee",
      subsets: ["latin", "latin-ext"],
      version: "v22",
      category: "sans-serif",
      variants: ["regular", "italic"],
    },
    {
      family: "Abel",
      subsets: ["latin"],
      version: "v18",
      category: "sans-serif",
      variants: ["regular"],
    },
  ] as Array<FontItem>;

  const tree = renderer
    .create(
      <Formik
        validate={(values: any) => {
          const errors = {} as any;

          if (!values?.firstName) {
            errors.firstName = "Required.";
          }

          return errors;
        }}
        initialValues={{}}
        onSubmit={() => {}}
      >
        <CustomSelect
          id={fieldName}
          name={fieldName}
          label={labelName}
          placeholder={labelName}
          data={fonts.map((x, index) => ({
            label: x.family,
            value: `'${x.family}', ${x.category}`,
            key: index,
          }))}
        />
      </Formik>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
