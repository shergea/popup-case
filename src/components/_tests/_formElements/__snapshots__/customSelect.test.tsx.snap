// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="mt-4"
>
  <label
    className="block mb-2 text-m font-medium text-gray-900 dark:text-gray-300"
    htmlFor="firstName"
  >
    First Name
  </label>
  <select
    className="border border-gray-300 text-gray-900 text-m rounded-lg block w-full p-2.5 outline-0"
    id="firstName"
    name="firstName"
    onBlur={[Function]}
    onChange={[Function]}
  >
    <option
      value=""
    >
      First Name
    </option>
    <option
      value="'ABeeZee', sans-serif"
    >
      ABeeZee
    </option>
    <option
      value="'Abel', sans-serif"
    >
      Abel
    </option>
  </select>
</div>
`;
